File: node-http-send/send-email.js
const API_URL = 'https://api.motorical.com/v1/send';

function requireEnv(name) {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Missing required environment variable: ${name}`);
  }
  return value;
}

async function sendEmail() {
  const apiKey = requireEnv('MOTORICAL_SMTP_API_KEY');
  const from = requireEnv('MOTORICAL_SMTP_FROM');
  const to = requireEnv('MOTORICAL_SMTP_TO');
  const dryRun = process.env.MOTORICAL_SMTP_DRY_RUN !== 'false';

  const response = await fetch(API_URL, {
    method: 'POST',
    headers: {
      Authorization: `ApiKey ${apiKey}`,
      'Content-Type': 'application/json',
      'Idempotency-Key': `example-${Date.now()}`
    },
    body: JSON.stringify({
      from,
      to: [to],
      subject: dryRun ? 'Motorical SMTP dry run' : 'Hello from Motorical SMTP',
      text: 'This message was sent with the Motorical SMTP HTTP Send API.',
      html: '<p>This message was sent with the <strong>Motorical SMTP</strong> HTTP Send API.</p>',
      dryRun
    })
  });

  const result = await response.json();
  if (!response.ok || !result.success) {
    throw new Error(result.error || `Motorical SMTP send failed with ${response.status}`);
  }

  console.log(JSON.stringify(result, null, 2));
}

sendEmail().catch((error) => {
  console.error(error.message);
  process.exitCode = 1;
});

File: python-http-send/send_email.py
import json
import os
import sys

import requests


API_URL = "https://api.motorical.com/v1/send"


def require_env(name):
    value = os.environ.get(name)
    if not value:
        raise RuntimeError(f"Missing required environment variable: {name}")
    return value


def main():
    api_key = require_env("MOTORICAL_SMTP_API_KEY")
    dry_run = os.environ.get("MOTORICAL_SMTP_DRY_RUN", "true").lower() != "false"

    response = requests.post(
        API_URL,
        headers={
            "Authorization": f"ApiKey {api_key}",
            "Content-Type": "application/json",
            "Idempotency-Key": f"python-example-{os.getpid()}",
        },
        json={
            "from": require_env("MOTORICAL_SMTP_FROM"),
            "to": [require_env("MOTORICAL_SMTP_TO")],
            "subject": "Motorical SMTP dry run" if dry_run else "Hello from Motorical SMTP",
            "text": "This message was sent with the Motorical SMTP HTTP Send API.",
            "html": "<p>This message was sent with the <strong>Motorical SMTP</strong> HTTP Send API.</p>",
            "dryRun": dry_run,
        },
        timeout=30,
    )

    result = response.json()
    if not response.ok or not result.get("success"):
        raise RuntimeError(result.get("error") or f"Motorical SMTP send failed with {response.status_code}")

    print(json.dumps(result, indent=2))


if __name__ == "__main__":
    try:
        main()
    except Exception as exc:
        print(str(exc), file=sys.stderr)
        sys.exit(1)
